test/spec/EditSpec.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module EditSpec where
+
+import System.Directory
+import Test.Hspec
+import Test.Mockery.Directory
+import Test.Mockery.Environment (withModifiedEnvironment)
+import TestUtils
+
+spec :: Spec
+spec =
+  describe "edit" $ around onTestFailureLogger $ do
+    repoDir <- runIO getCurrentDirectory
+    around_
+      ( withModifiedEnvironment [("NIX_CONFIG", "experimental-features =")]
+          . inTempDirectory
+      )
+      $ do
+        it "starts VSCode with the deno extension" $ \onTestFailureLog -> do
+          output <- runGarn ["edit", "--", "--list-extensions"] "" repoDir Nothing
+          onTestFailureLog output
+          stdout output `shouldBe` "denoland.vscode-deno\n"
```

Comment:
Maybe add a test that `garn edit` works even when the `garn.ts` file has type errors? (Spoiler, it doesn't.)

Reply:
Very nice catch!